### PR TITLE
Fix: Nested Connection On Cypher Field

### LINKED ISSUE #630: "Variable `...Connection` not defined"
Using package version: 2.5.0.

I have a schema that looks something like this:
```graphql
    workers: [User!]!
      @cypher(
        statement: """
        MATCH p = (this)-[*]->(u)<-[:HAS_REVIEWED]-()
        RETURN DISTINCT u
        """
      )
    reviewers: [User!]! @relationship(type: "HAS_REVIEWED", properties: "Review", direction: IN)
```

If I run the following query, it's successful:
```graphql
query Users($where: UserWhere) {
  users(where: $where) {
    workers {
      id
    }
    reviewersConnection {
      totalCount
    }
  }
}
```

If I run the following query, it fails:
```graphql
query Users($where: UserWhere) {
  users(where: $where) {
    workers {
      id
      reviewersConnection {
        totalCount
      }
    }
  }
}
```

Error:
```
 "stacktrace": [
            "Neo4jError: Variable `reviewersConnection` not defined (line 4, column 83 (offset: 257))",
            "\"RETURN u\", {this: this, auth: $auth}, true) | this_workers { .id, .role, reviewersConnection }] } as this\"",
            "                                                                                   ^",
            "",
            "    at captureStacktrace (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/result.js:239:17)",
            "    at new Result (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/result.js:59:23)",
            "    at newCompletedResult (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/transaction.js:372:12)",
            "    at Object.run (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/transaction.js:226:20)",
            "    at Transaction.run (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/transaction.js:98:34)",
            "    at /Users/munyiri/source/service/node_modules/@neo4j/graphql/src/utils/execute.ts:105:35",
            "    at TransactionExecutor._safeExecuteTransactionWork (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:92:26)",
            "    at TransactionExecutor._executeTransactionInsidePromise (/Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:83:34)",
            "    at /Users/munyiri/source/service/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:40:19",
            "    at new Promise (<anonymous>)"
          ]
```

Is there something I'm doing wrong? the expectation is that the autogenerated `reviewersConnection` should work in the nested object.

### PR DESCRIPTION
**Pull request by dmoree**
# Description

Fixes #630. Added missing `inRelationshipProjection: true` for cypher reference node projection.

# Issue

- #630

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

File: packages/graphql/tests/integration/directives/cypher.int.test.ts [hunk 1]
```diff
@@ -993,5 +994,109 @@ describe("cypher", () => {
                 );
             });
         });
+        // Reproduces https://github.com/neo4j/graphql/issues/630
```
Comment (danstarns) [inline packages/graphql/tests/integration/directives/cypher.int.test.ts:997]: This is great! One point for me, please could you put this test in the `/issues` folder?

Comment (dmoree) [reply packages/graphql/tests/integration/directives/cypher.int.test.ts]: Fixed in https://github.com/neo4j/graphql/pull/821/commits/82ecfac0ac87ab4505225d9c0ac09f5c62782323

**Comment by danstarns** [comment]
My apologies @dmoree I merged your branch incorrectly :/ We would like this PR to be part of a Hotfix and because we now use dev as the main branch we are introducing a Hotfix process stated here https://github.com/neo4j/graphql/pull/823.

Please could prepare this PR to go into master

**Comment by dmoree** [comment]
I'm not sure I follow. It looks as if the base branch has changed to `master`. I'm also not seeing any merge conflicts when merging with `master` as it says everything is up to date.

**Comment by darrellwarde** [comment]
Well I've royally messed this up, `git rebase` is not my friend. My apologies @dmoree, I'm hoping you have these changes stored locally!

Essentially, we need to release this off `master`, not `dev`, because we want to do so ASAP. However, purely changing the base in the GitHub UI would have meant merging all of the changes from `dev` into `master` which would have been very bad.

I tried to rebase from `dev` to `master`... As you can see this went terribly wrong. In hindsight, I should have made a new branch off `master` and cherry picked your commits into it.

My apologies again @dmoree, I hope this is enough of an explanation as to what we're trying to achieve?

**Pull request closed without merging**